pkg/issuer/acme/setup.go

```diff
@@ -69,47 +69,44 @@ const (
 
 // Setup will verify an existing ACME registration, or create one if not
 // already registered.
-func (a *Acme) Setup(ctx context.Context) error {
+func (a *Acme) Setup(ctx context.Context, issuer v1.GenericIssuer) error {
 	log := logf.FromContext(ctx)
 
 	// Correct reason and message for issuer's Ready condition must be always set
 	// before returning from this function. Status must be set if not false.
 	status := cmmeta.ConditionFalse
 	var reason, msg string
 	defer func() {
-		apiutil.SetIssuerCondition(a.issuer,
-			a.issuer.GetGeneration(),
+		apiutil.SetIssuerCondition(issuer,
+			issuer.GetGeneration(),
 			v1.IssuerConditionReady,
 			status,
 			reason,
 			msg)
 	}()
 
 	// check if user has specified a v1 account URL, and set a status condition if so.
-	if newURL, ok := acmev1ToV2Mappings[a.issuer.GetSpec().ACME.Server]; ok {
+	if newURL, ok := acmev1ToV2Mappings[issuer.GetSpec().ACME.Server]; ok {
 		reason = errorInvalidConfig
-		msg = fmt.Sprintf(messageTemplateUpdateToV2, a.issuer.GetSpec().ACME.Server, newURL)
+		msg = fmt.Sprintf(messageTemplateUpdateToV2, issuer.GetSpec().ACME.Server, newURL)
 		// Return nil, because we do not want to re-queue an Issuer with an invalid spec.
 		return nil
 	}
 
 	// if the namespace field is not set, we are working on a ClusterIssuer resource
 	// therefore we should check for the ACME private key in the 'cluster resource namespace'.
-	ns := a.issuer.GetObjectMeta().Namespace
-	if ns == "" {
-		ns = a.clusterResourceNamespace
-	}
+	ns := a.resourceNamespace(issuer)
```

Comment:
This is the only issuer that was not using `ctx.IssuerOptions.ResourceNamespace` yet.